src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,247 @@
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  SkyCountryData
+} from './types';
+
+import {
+  SkyPhoneFieldAdapterService
+} from './phone-field-adapter.service';
+
+import {
+  SkyPhoneFieldComponent
+} from './phone-field.component';
+
+// tslint:disable:no-forward-ref no-use-before-declare
+const SKY_PHONE_FIELD_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+
+const SKY_PHONE_FIELD_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+// tslint:enable
+
+@Directive({
+  selector: '[skyPhoneFieldInput]',
+  providers: [
+    SKY_PHONE_FIELD_VALUE_ACCESSOR,
+    SKY_PHONE_FIELD_VALIDATOR,
+    SkyPhoneFieldAdapterService
+  ]
+})
+export class SkyPhoneFieldInputDirective implements OnInit, OnDestroy, AfterViewInit,
+  ControlValueAccessor, Validator {
+
+  /**
+   * Disabling linting here as per the Angular style guide "Style 05-13" it is acceptable
+   * to alias when when the directive name is also an input property,
+   * and the directive name doesn't describe the property.
+   */
+  // tslint:disable-next-line:no-input-rename
+  @Input('skyPhoneFieldInput')
+  public skyPhoneFieldComponent: SkyPhoneFieldComponent;
+
+  @Input()
+  public set defaultCountry(value: string) {
+    this.skyPhoneFieldComponent.setDefaultCountry(value);
+    this._defaultCountryCode = value;
+  }
+
+  public get defaultCountry() {
+    return this._defaultCountryCode;
+  }
+
+  @Input()
+  public set disabled(value: boolean) {
+    this.skyPhoneFieldComponent.disabled = value;
+    this.adapterService.setElementDisabledState(this.elRef.nativeElement, value);
+    this._disabled = value;
+  }
+
+  public get disabled(): boolean {
+    return this._disabled || false;
+  }
+
+  @Input()
+  public formatModel: boolean = true;
+
+  @Input()
+  public skyPhoneFieldNoValidate: boolean = false;
+
+  @Output()
+  public selectedCountryChange = new EventEmitter<SkyCountryData>();
+
+  private set modelValue(value: string) {
+    this._modelValue = value;
+    this.adapterService.setElementValue(this.elRef.nativeElement, value);
+    if (value) {
+      let formattedValue = value;
+      if (this.formatModel) {
+        formattedValue = this.skyPhoneFieldComponent.formatNumber(value.toString());
+      }
+      this._onChange('+' + this.skyPhoneFieldComponent.selectedCountry.dialCode + ' ' + formattedValue);
+    } else {
+      this._onChange(value);
+    }
+    this._validatorChange();
+  }
+
+  private get modelValue(): string {
+    return this._modelValue;
+  }
+
+  private ngUnsubscribe = new Subject();
+
+  private _defaultCountryCode: string;
+
+  private _disabled: boolean;
+
+  private _modelValue: string;
+
+  public constructor(
+    private adapterService: SkyPhoneFieldAdapterService,
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private injector: Injector
+  ) { }
```

Comment:
Instead of renaming the directive `Input` (RE: `skyPhoneFieldInput`), you could instead inject the parent component into the constructor, and keep the `skyPhoneFieldInput` isolated to the directive's `selector` property. This will keep the API a bit cleaner (I want to do this with all of our other picker components in a breaking change).
```
@Directive({
  selector: '[skyPhoneFieldInput]'
})
// ...

// No longer needed:
// @Input()
// public skyPhoneFieldInput;

constructor (
  @Optional() private phoneFieldComponent: SkyPhoneFieldComponent
) { }
```

```
<sky-phone-field>
  <input type="text" skyPhoneFieldInput />
</sky-phone-field>
```

Here's an example of how we're doing something similar in Dropdown:
https://github.com/blackbaud/skyux-popovers/blob/master/src/app/public/modules/dropdown/dropdown-menu.component.ts#L81

Reply:
Done